docs/designs/anomaly_detection.md

```diff
@@ -0,0 +1,35 @@
+# Design Proposal for Anomaly Detection
+
+## Motivation
+Provide support for anomaly detection
+## Goals
+* Quickly identify and plot anomalies in time series data.
+* Create interactive visualizations.
+* Provide a helpful example notebook.
+
+## Non-Goals
+* Automatic visualizations (at the click of a button), such as GUI's. Proposed [here](https://github.com/data-describe/data-describe/blob/master/docs/designs/UI.md).
+* Creating notebooks for specific use cases i.e. sensor data, accelerometers, manufacturing.
+## UI or API
+* 2-D plot of time series data with confidence band. 
+* Use markers/colors to highlight anomalies.
+* Interactive plots for selecting time window.
+
+## Design
+There are multiple design aspects to be considered when creating this functionality.
+1. Supervised: Partitioning the data into train and test sets to calculate the confidence bands and error rates.
+2. Unsupervised: Fitting a model on the entire data, i.e. HDBSCAN and isolation forests
+3. Statistical methods: i.e. standard deviation from rolling mean and SH-ESD.
+```python
+import data_describe as dd
+dd.anomaly_detection(df) #Autodetect datatype for classification, regression, timeseries
```

Comment:
Can/will this share an API with general anomaly detection that isn't time series? How would the API change for one "mode" versus another? What things are configurable?